src/backend/InvenTree/plugin/registry.py

```diff
@@ -510,6 +513,17 @@ def safe_reference(plugin, key: str, active: bool = True):
             # Initialize package - we can be sure that an admin has activated the plugin
             logger.debug('Loading plugin `%s`', plg_name)
 
+            # If this is a third-party plugin, reload the source module
+            # This is required to ensure that separate processes are using the same code
+            if not builtin and not sample:
+                if plugin_module := sys.modules.get(plugin.__module__):
+                    logger.debug('Reloading plugin `%s`', plg_name)
```

Comment:
Is there a way to enhance this with information about the environment (ie if it is a worker)